src/ripple/app/main/Application.cpp

```diff
@@ -1105,7 +1105,12 @@ bool ApplicationImp::setup()
     m_overlay->setupValidatorKeyManifests (*config_, getWalletDB ());
 
     {
-        auto setup = setup_ServerHandler(*config_, std::cerr);
+        std::stringstream parseErr;
+        auto setup = setup_ServerHandler(*config_, parseErr);
```

Comment:
Potential problem here: `setup_ServerHandler` calls `parse_Ports` which can `throw`. The existing code would write prior to throwing, and this won't.